src/plugins/footnotes/index.ts

```diff
@@ -146,24 +145,35 @@ const deleteFootnoteWidget =
         if (node.type === schema.nodes.footnote && pos) {
           const targetNode = tableElement ? tableElement.node : view.state.doc
           const inlineFootnotes = getInlineFootnotes(id, targetNode)
-
-          const nodeWithPos = findParentNodeClosestToPos(
+          const footnotesElement = findParentNodeClosestToPos(
+            tr.doc.resolve(pos),
+            (node) => node.type === schema.nodes.footnotes_element
+          )
+          const tableElementFooter = findParentNodeClosestToPos(
             tr.doc.resolve(pos),
-            (node) => node.type === schema.nodes.footnote
+            (node) => node.type === schema.nodes.table_element_footer
           )
-          if (nodeWithPos) {
-            const { pos: fnPos, node: fnNode } = nodeWithPos
-            view.dispatch(tr.delete(fnPos, fnPos + fnNode.nodeSize))
+          // remove table-element-footer if it has only one footnote
+          if (
+            footnotesElement?.node.childCount === 1 &&
+            tableElementFooter?.node.childCount === 1
+          ) {
+            const { pos: fnPos, node: fnNode } = tableElementFooter
+            tr.delete(fnPos, fnPos + fnNode.nodeSize + 1)
```

Comment:
I have noticed a side effect of this change, the cited footnotes are not being deleted when there are multiple cited footnotes. can you please check ?
![image](https://github.com/user-attachments/assets/ba73daa3-c9fa-4b4d-9e80-c24337a34917)

![image](https://github.com/user-attachments/assets/0551fd68-75ad-4bb2-8a51-ffb16e04da6a)